WordPress/Sniffs/WP/EnqueuedResourceParametersSniff.php

```diff
@@ -0,0 +1,155 @@
+ * @link    https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards
+ * @license https://opensource.org/licenses/MIT MIT
+ */
+
+namespace WordPress\Sniffs\WP;
+
+use WordPress\AbstractFunctionParameterSniff;
+use PHP_CodeSniffer_Tokens as Tokens;
+
+/**
+ * This checks the enqueued 4th and 5th parameters to make sure the version and in_footer are set.
+ *
+ * If a source ($src) value is passed, then version ($ver) needs to have non-falsy value.
+ * If a source ($src) value is passed a check for in footer ($in_footer), warn the user if the value is falsy.
+ *
+ * @link https://developer.wordpress.org/reference/functions/wp_register_script/
+ * @link https://developer.wordpress.org/reference/functions/wp_enqueue_script/
+ * @link https://developer.wordpress.org/reference/functions/wp_register_style/
+ * @link https://developer.wordpress.org/reference/functions/wp_enqueue_style/
+ *
+ * @package WPCS\WordPressCodingStandards
+ *
+ * @since 1.0.0
+ */
+class EnqueuedResourceParametersSniff extends AbstractFunctionParameterSniff {
+
+	/**
+	 * The group name for this group of functions.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'Enqueued';
+
+	/**
+	 * List of enqueued functions that need to be check to make sure
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var array <string function_name> => <bool true>
+	 */
+	protected $target_functions = array(
+		'wp_register_script' => true,
+		'wp_enqueue_script'  => true,
+		'wp_register_style'  => true,
+		'wp_enqueue_style'   => true,
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		// Check to see if a source ($src) is specified.
+		if ( ! isset( $parameters[2] ) ) {
+			return;
+		}
+
+		/*
+		 * Version Check: Check to make sure the version is set explicitly.
+		 */
+
+		if ( ! isset( $parameters[4] ) || 'null' === $parameters[4]['raw'] ) {
+			$this->phpcsFile->addError(
+				'Resource version not set in call to %s(). This means new versions of the resource will not always be loaded due to browser caching.',
+				$stackPtr,
+				'MissingVersion',
+				array( $matched_content )
+			);
+		} else {
+			// The version argument should have a non-false value.
+			if ( $this->is_falsy( $parameters[4]['start'], $parameters[4]['end'] ) ) {
+				$this->phpcsFile->addError(
+					'Version parameter is not explicitly set for %s; This means that the WordPress core version will be used which is not recommended for plugin or theme development.',
```

Comment:
Suggestion: `Version parameter is not explicitly set or has been set to an equivalent of "false" for %s`